debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/SignalBasedIncrementalSnapshotChangeEventSource.java

```diff
@@ -70,11 +75,20 @@ protected void emitWindowOpen() throws SQLException {
 
     @Override
     protected void emitWindowClose(Partition partition, OffsetContext offsetContext) throws SQLException {
-        jdbcConnection.prepareUpdate(signalWindowStatement, x -> {
-            LOGGER.trace("Emitting close window for chunk = '{}'", context.currentChunkId());
-            x.setString(1, context.currentChunkId() + "-close");
-            x.setString(2, CloseIncrementalSnapshotWindow.NAME);
-        });
-        jdbcConnection.commit();
+
+        String signalTableName = getSignalTableName(connectorConfig.getSignalingDataCollectionId());
+
+        WatermarkWindowCloser watermarkWindowCloser = getWatermarkWindowCloser(connectorConfig, jdbcConnection, signalTableName);
+
+        watermarkWindowCloser.closeWindows(partition, offsetContext, context.currentChunkId());
+    }
+
+    private WatermarkWindowCloser getWatermarkWindowCloser(CommonConnectorConfig connectorConfig, JdbcConnection jdbcConnection, String signalTable) {
```

Comment:
I wonder if the `WindowCloser` can be provided directly by `SnapshotWatermarkStrategy`?

Reply:
It can be an option but for now, since it only used here, maybe this approach can be fine. 

I feel the missing of `Dependency Injection` :disappointed_relieved: